### ISSUE TITLE
Fix scoop version

### ISSUE THREAD

**Issue by JanHuege** [Jabba-Team/jabba#48]
Installing via scoop on windows 11:

use and current do not have any effect. Downloading jdks works but is useless without the version manager functionality.

**Comment by patrick-mccourt** [comment]
Hi Jan, 

Thanks for your patience while I've looked into this, the issue seems to be down to how scoop installs jabba. Scoop grabs the binary from the github releases and places it in the relevant directory. 

This misses one step from our install script which activates the shell integration. 

I have had an issue raised to look at the install process and making it a little better and haven't had the time. Package managers like scoop won't want a scripted install in so I need to come up with another way of setting up jabba to ensure the install works. 

In the mean time, I can suggest you could do 1 of 2 things: 

1. Uninstall the scoop version and use the powershell script in this repo 

or 2: Keep your scoop install and do this step manually:

Create a jabba.ps1 file in :

`C:\Users\patri\scoop\apps\jabba\current\` with the contents:

```
$env:JABBA_HOME="C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current"

function jabba
{
    $fd3=$([System.IO.Path]::GetTempFileName())
    $command="& 'C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\bin\jabba.exe' $args --fd3 `"$fd3`""
    & { $env:JABBA_SHELL_INTEGRATION="ON"; Invoke-Expression $command }
    $fd3content=$(Get-Content $fd3)
    if ($fd3content) {
        $expression=$fd3content.replace("export ","`$env:").replace("unset ","Remove-Item env:") -join "`n"
        if (-not $expression -eq "") { Invoke-Expression $expression }
    }
    Remove-Item -Force $fd3
}
```

Add in to your powershell profile:

`if (Test-Path "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1") { . "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1" }`

That should get you most of the jabba functionality (minus the default issue in your other issue)

**Comment by Baneeishaque** [comment]
i come with this solution - https://github.com/Baneeishaque/Scoop-Apps/blob/master/bucket/jabba.json.
i don't know it is the best approach. i am open for suggestions.

**Comment by JanecekPetr** [comment]
> Package managers like scoop won't want a scripted install in

I don't think so, Scoop has post-install scripts pretty often. I think it's fair to just open a ticket in [Scoop's main bucket](https://github.com/ScoopInstaller/Main), explain what you're doing, and provide a PR.

**Comment by diefesson** [comment]
Have you considered distributing jabba in a .zip file containing any additional required files besides the executable and/or implementing a  `jabba shell <shell>` command that outputs the snippet that needs to be added to the shell profile?

**Comment by ravindUwU** [comment]
I think a `jabba shell`/`jabba env` subcommand as suggested in https://github.com/Jabba-Team/jabba/issues/48#issuecomment-2493883502 would be ideal!

[`fnm`](https://github.com/Schniz/fnm/blob/3ea9e4d9899c51d4a1874c11031b6555e510c979/README.md#shell-setup), for example, uses `fnm env --shell bash/zsh/fish/powershell/cmd` to print out a corresponding script. Setting up their shell integration for PowerShell is just a matter of adding `fnm env --shell powershell (...and any other flags) | Out-String | Invoke-Expression` to the `$PROFILE`.

Their scoop manifest mentions this as a "note", which is shown after install: https://github.com/ScoopInstaller/Main/blob/1331ed1f001f87e07bdf9f8946dc1e78db32912a/bucket/fnm.json#L6

![](https://github.com/user-attachments/assets/d85d5fb9-9959-4de9-884f-aa8283e20dd3)

**Comment by tianlongJ** [comment]
> Hi Jan,
> 
> Thanks for your patience while I've looked into this, the issue seems to be down to how scoop installs jabba. Scoop grabs the binary from the github releases and places it in the relevant directory.
> 
> This misses one step from our install script which activates the shell integration.
> 
> I have had an issue raised to look at the install process and making it a little better and haven't had the time. Package managers like scoop won't want a scripted install in so I need to come up with another way of setting up jabba to ensure the install works.
> 
> In the mean time, I can suggest you could do 1 of 2 things:
> 
> 1. Uninstall the scoop version and use the powershell script in this repo
> 
> or 2: Keep your scoop install and do this step manually:
> 
> Create a jabba.ps1 file in :
> 
> `C:\Users\patri\scoop\apps\jabba\current\` with the contents:
> 
> ```
> $env:JABBA_HOME="C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current"
> 
> function jabba
> {
>     $fd3=$([System.IO.Path]::GetTempFileName())
>     $command="& 'C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\bin\jabba.exe' $args --fd3 `"$fd3`""
>     & { $env:JABBA_SHELL_INTEGRATION="ON"; Invoke-Expression $command }
>     $fd3content=$(Get-Content $fd3)
>     if ($fd3content) {
>         $expression=$fd3content.replace("export ","`$env:").replace("unset ","Remove-Item env:") -join "`n"
>         if (-not $expression -eq "") { Invoke-Expression $expression }
>     }
>     Remove-Item -Force $fd3
> }
> ```
> 
> Add in to your powershell profile:
> 
> `if (Test-Path "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1") { . "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1" }`
> 
> That should get you most of the jabba functionality (minus the default issue in your other issue)



> Hi Jan,
> 
> Thanks for your patience while I've looked into this, the issue seems to be down to how scoop installs jabba. Scoop grabs the binary from the github releases and places it in the relevant directory.
> 
> This misses one step from our install script which activates the shell integration.
> 
> I have had an issue raised to look at the install process and making it a little better and haven't had the time. Package managers like scoop won't want a scripted install in so I need to come up with another way of setting up jabba to ensure the install works.
> 
> In the mean time, I can suggest you could do 1 of 2 things:
> 
> 1. Uninstall the scoop version and use the powershell script in this repo
> 
> or 2: Keep your scoop install and do this step manually:
> 
> Create a jabba.ps1 file in :
> 
> `C:\Users\patri\scoop\apps\jabba\current\` with the contents:
> 
> ```
> $env:JABBA_HOME="C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current"
> 
> function jabba
> {
>     $fd3=$([System.IO.Path]::GetTempFileName())
>     $command="& 'C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\bin\jabba.exe' $args --fd3 `"$fd3`""
>     & { $env:JABBA_SHELL_INTEGRATION="ON"; Invoke-Expression $command }
>     $fd3content=$(Get-Content $fd3)
>     if ($fd3content) {
>         $expression=$fd3content.replace("export ","`$env:").replace("unset ","Remove-Item env:") -join "`n"
>         if (-not $expression -eq "") { Invoke-Expression $expression }
>     }
>     Remove-Item -Force $fd3
> }
> ```
> 
> Add in to your powershell profile:
> 
> `if (Test-Path "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1") { . "C:\Users\$YOUR_USERNAME\scoop\apps\jabba\current\jabba.ps1" }`
> 
> That should get you most of the jabba functionality (minus the default issue in your other issue)

On Windows, installed through Scoop, using the Git Bash command line to execute commands, and using the ps1 method will not work. Take fnm as an example. Using bash, you can add eval "$(fnm env --use-on-cd --shell bash)"; using powershell, you can add "fnm env --use-on-cd --shell powershell | Out-String | Invoke-Expression". Hopefully there is a similar compatibility
